src/onnx/parse_conv_transpose.cpp

```diff
@@ -122,11 +115,11 @@ struct parse_deconvolution : op_parser<parse_deconvolution>
         recalc_conv_attributes(values, kdims);
 
         op.from_value(values);
-        auto l1                   = info.add_instruction(op, l0, args[1]);
-        std::vector<int64_t> dims = to_int64_vector(l1->get_shape().lens());
-        std::vector<int64_t> curr_shape(dims.begin() + 2, dims.end());
+        auto l1 = info.add_instruction(op, l0, args[1]);
         if(asym_padding)
         {
+            std::vector<int64_t> dims = to_int64_vector(l1->get_shape().lens());
```

Comment:
I don't understand how this `if` condition part works. If it is `asym_padding` that means it must be static shape. What is it trying to do by slicing ? 

From what i gather, even though it is Asym padding,  it would compute convolution assuming sym padding because it assumes `padding_L == padding_R` inside MIGraphX's convolution backwards. 

But then how `slice` works here ?

Reply:
Passes symmetrical padding then does slicing off the end. Will be changed in another PR as discussed.